### ISSUE TITLE
Add a method to get (global space) AABBs of collision shapes

### ISSUE THREAD

**Issue by nenad-jalsovec** [godotengine/godot-proposals#3453]
### Describe the project you are working on

The issue arose in a discussion held over at godotforums.org. 
I was helping a user in building a simple buoyancy simulation system. Knowing world space AABBs was needed to determine submersion degree of physics bodies. We ended up using **VisualInstance.get_transformed_aabb()** on meshes that coincide with bodies. In the after-discussion I concluded that it'd be useful to have such function for collision shapes too.
Admins then gently pointed me here :)

### Describe the problem or limitation you are having in your project

Inability to easily determine axis aligned bounding boxes of transformed physics body collision shapes 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Adding CollisionShape.get_transformed_aabb() method into CollisionShape base class, analogous to such method in VisualInstance base class.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Simply:
`var bbox: AABB = my_collider.shape.get_transformed_aabb()`

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Not without including additional nodes into scene setup.

### Is there a reason why this should be core and not an add-on in the asset library?

- It's easy to implement. Imho, practically just a matter of exposing Bullet function **btCollisionShape::getAabb()** via Godot api
- It's counterpart to **VisualInstance.get_transformed_aabb()**

**Comment by pouleyKetchoupp** [comment]
Yes, this could be useful and it wouldn't be too hard to implement in godot physics either.

This is the internal function call in Godot physics 3D (and there's an equivalent one in 2D):
https://github.com/godotengine/godot/blob/fad7034e8b8275ce89424185c6ccee4de6aebd7f/servers/physics_3d/godot_collision_object_3d.h#L134

**Comment by JohnMeadow1** [comment]
Just spent 3 days trying to get AABB calculated for camera space, to get accurate Bounding Box.
Would love to see an option to get AABB of geometry transformed to coordinates other than local.
This would be helpful to get something like unit selection.
Finally I did this in GDscript
https://user-images.githubusercontent.com/23294139/231265441-5bcbb243-d9e4-4747-a1a3-33a64354c333.mp4
Green is BBox from AABB and it gives objects extra margins.
Yellow is BBox precise to geometry and about x1000 slower. 1.2 msec.

**Comment by DanielKinsman** [comment]
`get_transformed_aabb()` is gone from `VisualInstance3D` in 4.x so this proposal would be handy for more than just collision shapes now.

**Comment by Jamsers** [comment]
Currently there isn't a way to get the size of a physics object right? Can also be useful for, for example, making a physics gun and knowing how far away from the player to hold the physics object

**Comment by terrehbyte** [comment]
> Currently there isn't a way to get the size of a physics object right? Can also be useful for, for example, making a physics gun and knowing how far away from the player to hold the physics object

I'm working on this exact problem at the moment and am facing the same need. 😅

I'm not familiar with Godot's engine code yet, but at a glance, it seems like the 4.X equivalent for CollisionObject3D is here: [GodotCollisionObject3D::get_shape_aabb](https://github.com/godotengine/godot/blob/5b52b4b5c4f7c1232a6a814f65fce4c48f0f424b/modules/godot_physics_3d/godot_collision_object_3d.h#L135). I might try to write a small GDExtension module to provide a utility function to access it myself. Assuming it's "easy" to expose it to GDScript, I might look at a creating a PR to expose it in the engine proper.

I'm unsure if the incoming ability to switch between Godot Physics and Jolt Physics changes the API that I need to use though.

**Comment by Jamsers** [comment]
> I'm unsure if the incoming ability to switch between Godot Physics and Jolt Physics changes the API that I need to use though.

I'm pretty sure it doesn't, or at least the design intent is that it shouldn't.
